### DISCUSSION TITLE
Creating a ChatBot engine using Elsa

### DISCUSSION THREAD

**Discussion by vgb1993** [elsa-workflows/elsa-core#3573] [Ideas]
Hello there, 

Out of curosity, has anyone build a chatbot engine using Elsa?

With some custom activities and some triggers I think it's kind of easy. Can anyone spot any possible downsides? 

Big up Elsa!

**Comment by sfmskywalker** [comment]
Hi @vgb1993 ,

Not that I know of, and should be a lot of fun to do!

In case you're just experimenting, maybe you'd be interested in trying a few things with the new V3 bits.
Disclaimer: it's not yet complete and a lot of details are still missing (not to mention documentation), but at least the API is more polished than that of V2 and the designer is more advanced already.

https://v3.elsaworkflows.io/

**Reply by Vaughan-mci** [reply]
Surprised this thread hasn't gained more attention.

Unless I'm missing something, the path for an approvals workflow via a Slack or MS Teams channel would be done through a chatbot and is a major step up from the email route.

Definitely need to set some time aside to start experimenting with v3...

**Reply by sfmskywalker** [reply]
Exactly, such an integration would be so good. And easy to do since it's very easy to create custom activities and interact with APIs and webhooks. After the 3.0 release, I'm going to want to focus on providing real world examples such as these.
